R/confirmatoryfactoranalysis.R

```diff
@@ -425,11 +433,33 @@ confirmatoryFactorAnalysis <- function(jaspResults, dataset, options, ...) {
     maintab[["df"]]     <- c(           "."           ,           "."          )
     maintab[["pvalue"]] <- c(           "."           ,           "."          )
   } else {
-    fm <- lavaan::fitMeasures(cfaResult[["lav"]])
-    maintab[["mod"]]    <- c(gettext("Baseline model"), gettext("Factor model"))
-    maintab[["chisq"]]  <- fm[c("baseline.chisq", "chisq")]
-    maintab[["df"]]     <- fm[c("baseline.df", "df")]
-    maintab[["pvalue"]] <- c(NA, fm["pvalue"])
+    # check the fit options
+    fitOptions <- lavaan::inspect(cfaResult[["lav"]], what = "options")
+
+    # get the model fit
+    fitMeasures <- lavaan::fitmeasures(cfaResult[["lav"]])
+
+    if (cfaResult[["orderedVariables"]] && options[["estimator"]] == "default") {
+      # when the estimator is not default lavaan does not use the robust test
+      maintab[["mod"]]    <- c(gettext("Baseline model"), gettext("Factor model"))
+      maintab[["chisq"]]  <- fitMeasures[c("baseline.chisq.scaled", "chisq.scaled")]
+      maintab[["df"]]     <- fitMeasures[c("baseline.df.scaled", "df.scaled")]
+      maintab[["pvalue"]] <- c(NA, fitMeasures["pvalue.scaled"])
+      footnote <- gettextf("The estimator is %s and the test statistic is %s because there are categorical variables in the data.",
```

Comment:
please use positional arguments in `gettextf` (see https://github.com/jasp-stats/INTERNAL-jasp/issues/2125)